average.js

```diff
@@ -1,6 +1,24 @@
-
-function average(numbers) {
-  return numbers.reduce((p, c)=> p + c, 0) / numbers.length;
+function average(numbers){
+    var avg = 0;
+    if(numbers.length == 0){
+        return NaN;
+    }
+    else{
+        var sum = 0
+        // variable to check number of nan values 
+        var noOfNaN= 0
+        numbers.forEach(function(number){
+            if(!isNaN(number)){
+                sum += number;
+            }
+            else{
+                noOfNaN += 1
+            }
+        });
+        /* if nan values are present, 
+           then we need to subtract them from the length of the array */
+        avg = sum / (numbers.length-noOfNaN); 
+    }
+    return avg;
 }
-
 module.exports = {average};
```

Comment:
Try a new test: The tests in `test/average.test.js` express the functionality of the `average` function.
Consider this situation: In IoT applications, analog sensors will report occasional outliers. 'Outliers' are values so small or so large, they aren't expected to happen. We need new functionality to ignore these outliers.
First add a unit-test in `test/average.test.js` to ignore outliers. Name the test appropriately. Then implement it in the code.
You'll need thresholds to recognize outliers: Think of sensing battery-temperature values and determine reasonable values.

Reply:
added the test and the respective function.